helper/schema/provider.go

```diff
@@ -240,6 +192,8 @@ func (p *Provider) ValidateResource(
 }
 
 // Configure implementation of terraform.ResourceProvider interface.
+// This method should likely have Context pulled through it
+// However it will involve breaking the terraform.ResourceProvider interface
```

Comment:
I think this needs `Context` as it corresponds directly to a protocol rpc, we should just see what we can do about the interface.